File: Swift/long-pressed-name.swift
class Solution {
    func isLongPressedName(_ name: String, _ typed: String) -> Bool {
        var nameChars = Array(name)
        var typedChars = Array(typed)
        var i = 0

        for j in 0..<typedChars.count {
            if i < nameChars.count && nameChars[i] == typedChars[j] {
                i += 1
            } else if j == 0 || typedChars[j] != typedChars[j - 1] {
                return false
            }
        }

        return i == nameChars.count
    }
}

File: Python/long-pressed-name.py
class Solution:
    def isLongPressedName(self, name: 'str', typed: 'str') -> 'bool':
        """
        :type name: str
        :type typed: str
        :rtype: bool
        """
        i = 0

        for j in range(len(typed)):
            if i < len(name) and name[i] == typed[j]:
                i += 1
            elif j == 0 or typed[j] != typed[j - 1]:
                return False

        return i == len(name)
